AgentDiscoveries-Frontend/app/src/components/login.js

```diff
@@ -0,0 +1,123 @@
+import * as React from "react";
+import {
+    Form,
+    FormGroup,
+    FormControl,
+    Button,
+    ControlLabel
+} from "react-bootstrap";
+import * as CRUD from "./crud"
+
+export default class Login extends React.Component {
+
+    constructor() {
+        super();
+        this.state = {
+            authenticationMessage: "",
+            isLoggedInMessage: "",
+        }
+    }
+
+    componentDidMount() {
+        this.updateIsLoggedIn();
+    }
+
+    render() {
+        return (
+            <div>
+                <div>{this.state.isLoggedInMessage}</div>
+                <Button onClick={this.logOut.bind(this)}>Log out</Button>
+                
+                <Form onSubmit={this.handleLogIn.bind(this)}>
+                    <h3>Sign in</h3>
+                    <ControlLabel bsStyle="warning">{this.state.authenticationMessage}</ControlLabel> 
+                    <FormGroup>
+                        <FormControl type="text" inputRef={username => this.username = username} placeholder="enter your username" />
+                        <FormControl type="password" inputRef={password => this.password = password} placeholder="enter password" />
+                        <Button type="submit">Login</Button><Button onClick={this.handleRegister.bind(this)}>Register</Button>
+                    </FormGroup>
+                </Form>
+            </div>
+        );
+    }
+
+    
+
+    handleLogIn(e) {
+        e.preventDefault();
+        var requestBodyJSON = {
+            "username": this.username.value,
+            "password": this.password.value
+        }
+
+        this.makeAuthenticationAPICall("/v1/token", requestBodyJSON)
+            .then(response => response.json())
+            .then(response => {
+                let token = response.token;
+                window.localStorage.setItem("Token", token);
+                this.updateIsLoggedIn();
+                if (!response.errorCode){
+                    this.setState({authenticationMessage: "Signed in successfully"});
+                } else {
+                    this.setState({authenticationMessage: "response.message"});
+                }
+            })
+            .catch(err => {
+                this.setState({ authenticationMessage: err });
+            });
+
+    }
+
+    handleRegister(e) {
+        e.preventDefault();
+        var requestBodyJSON = {
+            "username": this.username.value,
+            "password": this.password.value
+        }
+
+        this.makeAuthenticationAPICall("/v1/makeuser", requestBodyJSON)

```

Comment:
How hard would it be to get this to log you in as well? (Perhaps something to look at after this is merged. Quite keen to get this stuff on master)